src/app/rpmostree-compose-builtin-tree.c

```diff
@@ -118,6 +120,47 @@ compute_checksum_from_treefile_and_goal (RpmOstreeTreeComposeContext   *self,
     g_checksum_update (checksum, buf, len);
   }
 
+  if (add_files)
+    {
+      guint i, len = json_array_get_length (add_files);
+      for (i = 0; i < len; i++)
+        {
+          gs_unref_object GFile *srcfile = NULL;
+          const char *src, *dest;
+          JsonArray *add_el = json_array_get_array_element (add_files, i);
+          gs_unref_object GFile *child = NULL;
+
+          if (!add_el)
+            {
+              g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                           "Element in add-files is not an array");
+              goto out;
+            }
+          src = _rpmostree_jsonutil_array_require_string_element (add_el, 0, error);
+          if (!src)
+            goto out;
+
+          dest = _rpmostree_jsonutil_array_require_string_element (add_el, 1, error);
+          if (!dest)
+            goto out;
+
+          srcfile = g_file_resolve_relative_path (contextdir, src);
+
+          {
+            g_autofree guchar *ret_csum = NULL;
+            if (!ostree_checksum_file (srcfile,
```

Comment:
See my comment in https://github.com/ostreedev/ostree/pull/219#discussion_r57238291 - this will also checksum uid/gid/xattrs, which is a bit weird since we'll be checksumming e.g. `security.selinux` on the host system, but on target may have a different one.

This does raise a general question about what the uid/gid for copied files should be.  If for example rpm-ostree starts supporting non-root operation, we'd still want the target files to be owned by UID `0`, not whatever UID was used to build the system.